host/resource/resource.go

```diff
@@ -62,3 +65,43 @@ func SetDefaults(r *Resources) {
 		(*r)[typ] = spec
 	}
 }
+
+func ToType(s string) (Type, bool) {
+	for typ := range defaults {
+		if string(typ) == s {
+			return typ, true
+		}
+	}
+	return Type(""), false
+}
+
+func ParseLimit(typ Type, s string) (int64, error) {
+	switch typ {
+	case TypeMemory:
+		return units.RAMInBytes(s)
+	default:
+		return units.FromHumanSize(s)
+	}
+}
+
+func FormatLimit(typ Type, limit int64) string {
+	switch typ {
+	case TypeMemory:
+		return byteSize(limit)
+	default:
+		return strconv.FormatInt(limit, 10)
+	}
+}
+
+var byteUnits = []string{"B", "kB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"}
```

Comment:
I'd like for us to add support for Novabytes (1000^9 / 10^27) and Dexabytes (1000^10 / 10^30) to help promote the new names I've devised and support higher limits on more powerful nodes.